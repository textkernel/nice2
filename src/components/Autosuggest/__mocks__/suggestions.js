export const SUGGESTIONS = [
    { name: 'Sun' },
    { name: 'Moon' },
    { name: 'Mercury' },
    { name: 'Venus' },
    { name: 'Earth' },
    { name: 'Mars' },
    { name: 'Jupiter' },
    { name: 'Saturn' },
    { name: 'Neptun' }
];

export const SUGGESTION_TO_STRING = item => (item ? item.name : '');
