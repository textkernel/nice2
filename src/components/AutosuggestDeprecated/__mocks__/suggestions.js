export const SUGGESTIONS = [
    { name: 'Sun' },
    { name: 'Moon' },
    { name: 'Mercury' },
    { name: 'Venus' },
    { name: 'Earth' },
    { name: 'Mars' },
    { name: 'Jupiter' },
    { name: 'Saturn' },
    { name: 'Neptun' },
];

export const COMPLEX_SUGGESTIONS = [
    { name: 'Sun', type: 'Star', count: 1234 },
    { name: 'Moon', type: 'Star', count: 1234 },
    { name: 'Mercury', type: 'Planet', count: 1234 },
    { name: 'Venus', type: 'Planet', count: 1234 },
    { name: 'Earth', type: 'Planet', count: 1234 },
    { name: 'Mars', type: 'Planet', count: 1234 },
    { name: 'Jupiter', type: 'Planet', count: 1234 },
    { name: 'Saturn', type: 'Planet', count: 1234 },
    { name: 'Neptun', type: 'Planet', count: 1234 },
];

export const SUGGESTION_TO_STRING = (item) => (item ? item.name : '');
