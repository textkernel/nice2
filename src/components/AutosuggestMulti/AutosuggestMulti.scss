.AutosuggestMulti {
    &__wrapper {
        cursor: pointer;
        display: flex;
        flex-wrap: nowrap;
        flex-direction: row;
        align-items: stretch;
        overflow: hidden;
        height: 36px;
        box-sizing: border-box;
        outline: none;
        padding: var(--spacing-normal);
        padding-top: var(--margin);

        &--isFocused {
            min-height: 36px;
            height: auto;
            flex-wrap: wrap;
        }
    }

    &__input {
        border: 0;
        margin: var(--spacing-normal);
        margin-top: 4px;
        flex: 1;
        font-size: var(--font-size-normal);
        min-width: 100px;
        height: 22px;

        &:focus {
            outline: none;
        }

        &::placeholder {
            color: var(--color-muted);
        }
    }
}
