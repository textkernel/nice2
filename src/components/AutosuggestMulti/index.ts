export { AutosuggestMulti } from './AutosuggestMulti';
export { SuggestionTag } from './SuggestionTag';
