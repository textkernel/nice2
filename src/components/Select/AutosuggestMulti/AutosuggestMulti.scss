.AutosuggestMulti {
    &__wrapper {
        cursor: pointer;
        display: flex;
        flex-wrap: nowrap;
        flex-direction: row;
        align-items: stretch;
        overflow: hidden;
        height: 34px;
        box-sizing: border-box;
        outline: none;
        padding: var(--spacing-normal);
        padding-bottom: var(--margin);

        &--isFocused {
            min-height: 34px;
            height: auto;
            flex-wrap: wrap;
        }
    }

    &__input {
        border: 0;
        margin: var(--spacing-normal);
        padding-left: var(--spacing-normal);
        flex: 1;
        font-size: var(--font-size-normal);
        min-width: 105px;

        &--hidden {
            position: absolute;
            opacity: 0;
        }

        &:focus {
            outline: none;
        }

        &::placeholder {
            color: var(--color-muted);
        }
    }
}
