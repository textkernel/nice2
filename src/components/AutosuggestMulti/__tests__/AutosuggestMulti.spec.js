import React from 'react';
import toJson from 'enzyme-to-json';
import { AutosuggestMulti } from '../AutosuggestMulti';
import { SUGGESTIONS, SUGGESTION_TO_STRING } from '../../Autosuggest/__mocks__/suggestions';

describe('AutosuggestMulti', () => {
    const suggestionToString = SUGGESTION_TO_STRING;
    const inputPlaceholder = 'type here...';
    const mockOnSelectionChange = jest.fn();
    const mockOnInputValueChange = jest.fn();
    const mockOnBlur = jest.fn();

    let suggestionsList = [];
    let selectedSuggestions = [];

    let wrapper;
    const setFocusOnAutosuggest = () =>
        wrapper.find('[role="searchbox"]').simulate('keyDown', { key: 'Enter' });

    beforeEach(() => {
        wrapper = mount(
            <AutosuggestMulti
                selectedSuggestions={selectedSuggestions}
                suggestions={suggestionsList}
                suggestionToString={suggestionToString}
                inputPlaceholder={inputPlaceholder}
                onSelectionChange={mockOnSelectionChange}
                onInputValueChange={mockOnInputValueChange}
                onBlur={mockOnBlur}
            />
        );
    });

    afterEach(() => {
        suggestionsList = [];
        selectedSuggestions = [];
        wrapper.unmount();
    });

    describe('rendering', () => {
        it('should initially render empty component correctly', () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        it('should render empty component correctly when focused', async () => {
            setFocusOnAutosuggest();
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        it('should render tag for each selected selection when component is focused', () => {
            selectedSuggestions = SUGGESTIONS.slice(0, 5);
            wrapper.setProps({ selectedSuggestions });
            setFocusOnAutosuggest();

            expect(wrapper.find('SuggestionTag')).toHaveLength(selectedSuggestions.length);
        });
    });
    describe('focusing and blurring the search field', () => {
        it('should clear input value on pressing Escape button', () => {
            const textInputValue = 'driver';
            setFocusOnAutosuggest();
            wrapper.find('input').simulate('change', { target: { value: textInputValue } });
            expect(wrapper.find('input').props().value).toEqual(textInputValue);
            wrapper.find('input').simulate('keyDown', { key: 'Escape' });
            expect(wrapper.find('input').props().value).toEqual('');
        });
        it('should blur on pressing Tab button', () => {
            setFocusOnAutosuggest();
            wrapper.find('input').simulate('keyDown', { key: 'Tab' });
            expect(mockOnBlur).toHaveBeenCalled();
        });
    });
    describe('callbacks', () => {
        describe('onSelectionChange', () => {
            it('should be called on clicking on a suggestion', () => {
                suggestionsList = SUGGESTIONS.slice(1, 20);
                wrapper.setProps({ suggestions: suggestionsList });
                setFocusOnAutosuggest();

                expect(mockOnSelectionChange).not.toHaveBeenCalled();

                wrapper.find('input').simulate('change', { target: { value: 'a' } });

                expect(toJson(wrapper)).toMatchSnapshot();

                wrapper
                    .find('li')
                    .first()
                    .children()
                    .simulate('click');

                expect(mockOnSelectionChange).toHaveBeenCalled();
            });
            it('should be called also when clicking on a suggestion the second time in a row', () => {
                suggestionsList = SUGGESTIONS.slice(1, 20);
                wrapper.setProps({ suggestions: suggestionsList });
                setFocusOnAutosuggest();

                expect(mockOnSelectionChange).not.toHaveBeenCalled();

                wrapper
                    .find('li')
                    .first()
                    .children()
                    .simulate('click');

                expect(mockOnSelectionChange).toHaveBeenCalledTimes(1);

                setFocusOnAutosuggest();
                wrapper
                    .find('li')
                    .first()
                    .children()
                    .simulate('click');

                expect(mockOnSelectionChange).toHaveBeenCalledTimes(2);
            });
            it('should be called on deleting a suggestion by clicking on the x button next to it', () => {
                selectedSuggestions = SUGGESTIONS.slice(0, 5);
                wrapper.setProps({ selectedSuggestions });
                setFocusOnAutosuggest();

                expect(mockOnSelectionChange).not.toHaveBeenCalled();
                expect(wrapper.find('SuggestionTag')).toHaveLength(selectedSuggestions.length);

                wrapper
                    .find('SuggestionTag')
                    .at(2)
                    .find('button')
                    .simulate('click');

                expect(mockOnSelectionChange).toHaveBeenCalledWith(selectedSuggestions[2]);
            });
            it('should be called on deleting a suggestion by hitting backspace in the empty input field', () => {
                selectedSuggestions = SUGGESTIONS.slice(0, 5);
                wrapper.setProps({ selectedSuggestions });
                setFocusOnAutosuggest();

                expect(mockOnSelectionChange).not.toHaveBeenCalled();

                wrapper.find('input').simulate('keyDown', { key: 'Backspace' });

                expect(mockOnSelectionChange).toHaveBeenCalled();
            });
            it('should not be called on deleting a suggestion by hitting backspace in an input field with value', () => {
                selectedSuggestions = SUGGESTIONS.slice(0, 5);
                wrapper.setProps({ selectedSuggestions });
                setFocusOnAutosuggest();
                wrapper.find('input').simulate('change', { target: { value: 'driver' } });

                expect(mockOnSelectionChange).not.toHaveBeenCalled();

                wrapper.find('input').simulate('keyDown', { key: 'Backspace' });

                expect(mockOnSelectionChange).not.toHaveBeenCalled();
            });
        });
        it('should call onInputValueChange when typing into input field', () => {
            expect(mockOnInputValueChange).not.toHaveBeenCalled();

            wrapper.find('input').simulate('change', { target: { value: 'driver' } });

            expect(mockOnInputValueChange).toHaveBeenCalled();
        });
        it('should clear the input field when a suggestion was selected', () => {
            const textInputValue = 'driver';
            suggestionsList = SUGGESTIONS.slice(1, 20);
            wrapper.setProps({ suggestions: suggestionsList });
            wrapper.find('input').simulate('change', { target: { value: textInputValue } });
            setFocusOnAutosuggest();

            expect(wrapper.find('input').props().value).toEqual(textInputValue);

            wrapper
                .find('li')
                .first()
                .children()
                .simulate('click');

            expect(wrapper.find('input').props().value).toEqual('');
        });
    });
});
