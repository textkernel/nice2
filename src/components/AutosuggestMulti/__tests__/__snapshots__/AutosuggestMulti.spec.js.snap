// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutosuggestMulti callbacks onSelectionChange should be called on clicking on a suggestion 1`] = `
<AutosuggestMulti
  inputPlaceholder="type here..."
  onBlur={[MockFunction]}
  onInputValueChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "a",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onSelectionChange={[MockFunction]}
  selectedSuggestions={Array []}
  suggestionToString={[Function]}
  suggestions={
    Array [
      Object {
        "name": "Moon",
      },
      Object {
        "name": "Mercury",
      },
      Object {
        "name": "Venus",
      },
      Object {
        "name": "Earth",
      },
      Object {
        "name": "Mars",
      },
      Object {
        "name": "Jupiter",
      },
      Object {
        "name": "Saturn",
      },
      Object {
        "name": "Neptun",
      },
    ]
  }
>
  <SelectBase
    blurredRenderer={[Function]}
    clearInputAfterSelection={true}
    clearTitle=""
    focusedRenderer={[Function]}
    inputRef={
      Object {
        "current": <input
          aria-activedescendant="downshift-5-item-0"
          aria-autocomplete="list"
          aria-controls="downshift-5-menu"
          aria-labelledby="downshift-5-label"
          autocomplete="off"
          class="AutosuggestMulti__input"
          id="downshift-5-input"
          placeholder="type here..."
          value="a"
        />,
      }
    }
    keepExpandedAfterSelection={true}
    listRenderer={[Function]}
    onBlur={[MockFunction]}
    onInputValueChange={[Function]}
    onSelectionChange={[MockFunction]}
    showClearButton={false}
    suggestionToString={[Function]}
    suggestions={
      Array [
        Object {
          "name": "Moon",
        },
        Object {
          "name": "Mercury",
        },
        Object {
          "name": "Venus",
        },
        Object {
          "name": "Earth",
        },
        Object {
          "name": "Mars",
        },
        Object {
          "name": "Jupiter",
        },
        Object {
          "name": "Saturn",
        },
        Object {
          "name": "Neptun",
        },
      ]
    }
  >
    <div
      className="SelectBase"
    >
      <Downshift
        defaultHighlightedIndex={0}
        defaultIsOpen={false}
        environment={[Window]}
        getA11yStatusMessage={[Function]}
        inputValue="a"
        itemToString={[Function]}
        onChange={[Function]}
        onInputValueChange={[Function]}
        onOuterClick={[Function]}
        onSelect={[Function]}
        onStateChange={[Function]}
        onUserAction={[Function]}
        scrollIntoView={[Function]}
        selectedItemChanged={[Function]}
        stateReducer={[Function]}
        suppressRefError={false}
      >
        <div
          aria-expanded={true}
          aria-haspopup="listbox"
          aria-labelledby="downshift-5-label"
          aria-owns="downshift-5-menu"
          className="SelectBase__main SelectBase__main--focused"
          role="combobox"
        >
          <FieldWrapper
            className="SelectBase__field"
            clearLabel=""
            isFocused={true}
            onClear={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            showClearButton={false}
          >
            <div
              className="FieldWrapper FieldWrapper--isFocused SelectBase__field"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="searchbox"
              tabIndex={0}
            >
              <div
                className="FieldWrapper__content"
              >
                <div
                  className="AutosuggestMulti__wrapper AutosuggestMulti__wrapper--isFocused"
                >
                  <input
                    aria-activedescendant="downshift-5-item-0"
                    aria-autocomplete="list"
                    aria-controls="downshift-5-menu"
                    aria-labelledby="downshift-5-label"
                    autoComplete="off"
                    className="AutosuggestMulti__input"
                    id="downshift-5-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="type here..."
                    value="a"
                  />
                </div>
                <List
                  aria-labelledby="downshift-5-label"
                  className="SelectBase__list--focused"
                  doSelectOnNavigate={false}
                  id="downshift-5-menu"
                  isControlledNavigation={true}
                  isDivided={false}
                  role="listbox"
                >
                  <ul
                    aria-labelledby="downshift-5-label"
                    className="List SelectBase__list--focused"
                    id="downshift-5-menu"
                    role="listbox"
                  >
                    <SuggestionsList
                      className="List__item"
                      getItemProps={[Function]}
                      highlightedIndex={0}
                      inputValue="a"
                      key=".0"
                      noSuggestionsPlaceholder=""
                      suggestionToString={[Function]}
                      suggestions={
                        Array [
                          Object {
                            "name": "Moon",
                          },
                          Object {
                            "name": "Mercury",
                          },
                          Object {
                            "name": "Venus",
                          },
                          Object {
                            "name": "Earth",
                          },
                          Object {
                            "name": "Mars",
                          },
                          Object {
                            "name": "Jupiter",
                          },
                          Object {
                            "name": "Saturn",
                          },
                          Object {
                            "name": "Neptun",
                          },
                        ]
                      }
                      useOptimizeRender={false}
                    >
                      <ListItem
                        aria-selected={true}
                        disabled={false}
                        highlightContext="brand"
                        id="downshift-5-item-0"
                        isHighlighted={true}
                        isSelected={false}
                        key="Moon"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseMove={[Function]}
                        role="option"
                        style={Object {}}
                      >
                        <li
                          aria-selected={true}
                          className="ListItem ListItem--isHighlighted ListItem--highlightContext_brand ListItem--clickable"
                          id="downshift-5-item-0"
                          onMouseDown={[Function]}
                          onMouseMove={[Function]}
                          role="option"
                          style={Object {}}
                        >
                          <div
                            className="ListItem__container"
                            onClick={[Function]}
                            role="presentation"
                          >
                            <MarkedText
                              inline={true}
                              key=".0"
                              marker="a"
                            >
                              <Text
                                context="default"
                                inline={true}
                                size="normal"
                              >
                                <span>
                                  Moon
                                </span>
                              </Text>
                            </MarkedText>
                          </div>
                        </li>
                      </ListItem>
                      <ListItem
                        aria-selected={false}
                        disabled={false}
                        highlightContext="brand"
                        id="downshift-5-item-1"
                        isHighlighted={false}
                        isSelected={false}
                        key="Mercury"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseMove={[Function]}
                        role="option"
                        style={Object {}}
                      >
                        <li
                          aria-selected={false}
                          className="ListItem ListItem--highlightContext_brand ListItem--clickable"
                          id="downshift-5-item-1"
                          onMouseDown={[Function]}
                          onMouseMove={[Function]}
                          role="option"
                          style={Object {}}
                        >
                          <div
                            className="ListItem__container"
                            onClick={[Function]}
                            role="presentation"
                          >
                            <MarkedText
                              inline={true}
                              key=".0"
                              marker="a"
                            >
                              <Text
                                context="default"
                                inline={true}
                                size="normal"
                              >
                                <span>
                                  Mercury
                                </span>
                              </Text>
                            </MarkedText>
                          </div>
                        </li>
                      </ListItem>
                      <ListItem
                        aria-selected={false}
                        disabled={false}
                        highlightContext="brand"
                        id="downshift-5-item-2"
                        isHighlighted={false}
                        isSelected={false}
                        key="Venus"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseMove={[Function]}
                        role="option"
                        style={Object {}}
                      >
                        <li
                          aria-selected={false}
                          className="ListItem ListItem--highlightContext_brand ListItem--clickable"
                          id="downshift-5-item-2"
                          onMouseDown={[Function]}
                          onMouseMove={[Function]}
                          role="option"
                          style={Object {}}
                        >
                          <div
                            className="ListItem__container"
                            onClick={[Function]}
                            role="presentation"
                          >
                            <MarkedText
                              inline={true}
                              key=".0"
                              marker="a"
                            >
                              <Text
                                context="default"
                                inline={true}
                                size="normal"
                              >
                                <span>
                                  Venus
                                </span>
                              </Text>
                            </MarkedText>
                          </div>
                        </li>
                      </ListItem>
                      <ListItem
                        aria-selected={false}
                        disabled={false}
                        highlightContext="brand"
                        id="downshift-5-item-3"
                        isHighlighted={false}
                        isSelected={false}
                        key="Earth"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseMove={[Function]}
                        role="option"
                        style={Object {}}
                      >
                        <li
                          aria-selected={false}
                          className="ListItem ListItem--highlightContext_brand ListItem--clickable"
                          id="downshift-5-item-3"
                          onMouseDown={[Function]}
                          onMouseMove={[Function]}
                          role="option"
                          style={Object {}}
                        >
                          <div
                            className="ListItem__container"
                            onClick={[Function]}
                            role="presentation"
                          >
                            <MarkedText
                              inline={true}
                              key=".0"
                              marker="a"
                            >
                              <Text
                                context="default"
                                inline={true}
                                size="normal"
                              >
                                <span>
                                  E
                                  <mark
                                    className="MarkedText__marked"
                                    key="match1"
                                  >
                                    a
                                  </mark>
                                  rth
                                </span>
                              </Text>
                            </MarkedText>
                          </div>
                        </li>
                      </ListItem>
                      <ListItem
                        aria-selected={false}
                        disabled={false}
                        highlightContext="brand"
                        id="downshift-5-item-4"
                        isHighlighted={false}
                        isSelected={false}
                        key="Mars"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseMove={[Function]}
                        role="option"
                        style={Object {}}
                      >
                        <li
                          aria-selected={false}
                          className="ListItem ListItem--highlightContext_brand ListItem--clickable"
                          id="downshift-5-item-4"
                          onMouseDown={[Function]}
                          onMouseMove={[Function]}
                          role="option"
                          style={Object {}}
                        >
                          <div
                            className="ListItem__container"
                            onClick={[Function]}
                            role="presentation"
                          >
                            <MarkedText
                              inline={true}
                              key=".0"
                              marker="a"
                            >
                              <Text
                                context="default"
                                inline={true}
                                size="normal"
                              >
                                <span>
                                  M
                                  <mark
                                    className="MarkedText__marked"
                                    key="match1"
                                  >
                                    a
                                  </mark>
                                  rs
                                </span>
                              </Text>
                            </MarkedText>
                          </div>
                        </li>
                      </ListItem>
                      <ListItem
                        aria-selected={false}
                        disabled={false}
                        highlightContext="brand"
                        id="downshift-5-item-5"
                        isHighlighted={false}
                        isSelected={false}
                        key="Jupiter"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseMove={[Function]}
                        role="option"
                        style={Object {}}
                      >
                        <li
                          aria-selected={false}
                          className="ListItem ListItem--highlightContext_brand ListItem--clickable"
                          id="downshift-5-item-5"
                          onMouseDown={[Function]}
                          onMouseMove={[Function]}
                          role="option"
                          style={Object {}}
                        >
                          <div
                            className="ListItem__container"
                            onClick={[Function]}
                            role="presentation"
                          >
                            <MarkedText
                              inline={true}
                              key=".0"
                              marker="a"
                            >
                              <Text
                                context="default"
                                inline={true}
                                size="normal"
                              >
                                <span>
                                  Jupiter
                                </span>
                              </Text>
                            </MarkedText>
                          </div>
                        </li>
                      </ListItem>
                      <ListItem
                        aria-selected={false}
                        disabled={false}
                        highlightContext="brand"
                        id="downshift-5-item-6"
                        isHighlighted={false}
                        isSelected={false}
                        key="Saturn"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseMove={[Function]}
                        role="option"
                        style={Object {}}
                      >
                        <li
                          aria-selected={false}
                          className="ListItem ListItem--highlightContext_brand ListItem--clickable"
                          id="downshift-5-item-6"
                          onMouseDown={[Function]}
                          onMouseMove={[Function]}
                          role="option"
                          style={Object {}}
                        >
                          <div
                            className="ListItem__container"
                            onClick={[Function]}
                            role="presentation"
                          >
                            <MarkedText
                              inline={true}
                              key=".0"
                              marker="a"
                            >
                              <Text
                                context="default"
                                inline={true}
                                size="normal"
                              >
                                <span>
                                  S
                                  <mark
                                    className="MarkedText__marked"
                                    key="match1"
                                  >
                                    a
                                  </mark>
                                  turn
                                </span>
                              </Text>
                            </MarkedText>
                          </div>
                        </li>
                      </ListItem>
                      <ListItem
                        aria-selected={false}
                        disabled={false}
                        highlightContext="brand"
                        id="downshift-5-item-7"
                        isHighlighted={false}
                        isSelected={false}
                        key="Neptun"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseMove={[Function]}
                        role="option"
                        style={Object {}}
                      >
                        <li
                          aria-selected={false}
                          className="ListItem ListItem--highlightContext_brand ListItem--clickable"
                          id="downshift-5-item-7"
                          onMouseDown={[Function]}
                          onMouseMove={[Function]}
                          role="option"
                          style={Object {}}
                        >
                          <div
                            className="ListItem__container"
                            onClick={[Function]}
                            role="presentation"
                          >
                            <MarkedText
                              inline={true}
                              key=".0"
                              marker="a"
                            >
                              <Text
                                context="default"
                                inline={true}
                                size="normal"
                              >
                                <span>
                                  Neptun
                                </span>
                              </Text>
                            </MarkedText>
                          </div>
                        </li>
                      </ListItem>
                    </SuggestionsList>
                  </ul>
                </List>
              </div>
            </div>
          </FieldWrapper>
        </div>
      </Downshift>
    </div>
  </SelectBase>
</AutosuggestMulti>
`;

exports[`AutosuggestMulti rendering should initially render empty component correctly 1`] = `
<AutosuggestMulti
  inputPlaceholder="type here..."
  onBlur={[MockFunction]}
  onInputValueChange={[MockFunction]}
  onSelectionChange={[MockFunction]}
  selectedSuggestions={Array []}
  suggestionToString={[Function]}
  suggestions={Array []}
>
  <SelectBase
    blurredRenderer={[Function]}
    clearInputAfterSelection={true}
    clearTitle=""
    focusedRenderer={[Function]}
    inputRef={
      Object {
        "current": <input
          aria-autocomplete="list"
          aria-labelledby="downshift-0-label"
          autocomplete="off"
          class="AutosuggestMulti__input"
          id="downshift-0-input"
          placeholder="type here..."
          value=""
        />,
      }
    }
    keepExpandedAfterSelection={true}
    listRenderer={[Function]}
    onBlur={[MockFunction]}
    onInputValueChange={[Function]}
    onSelectionChange={[MockFunction]}
    showClearButton={false}
    suggestionToString={[Function]}
    suggestions={Array []}
  >
    <div
      className="SelectBase"
    >
      <Downshift
        defaultHighlightedIndex={0}
        defaultIsOpen={false}
        environment={[Window]}
        getA11yStatusMessage={[Function]}
        inputValue=""
        itemToString={[Function]}
        onChange={[Function]}
        onInputValueChange={[Function]}
        onOuterClick={[Function]}
        onSelect={[Function]}
        onStateChange={[Function]}
        onUserAction={[Function]}
        scrollIntoView={[Function]}
        selectedItemChanged={[Function]}
        stateReducer={[Function]}
        suppressRefError={false}
      >
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-labelledby="downshift-0-label"
          aria-owns={null}
          className="SelectBase__main"
          role="combobox"
        >
          <FieldWrapper
            className="SelectBase__field"
            clearLabel=""
            isFocused={false}
            onClear={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            showClearButton={false}
          >
            <div
              className="FieldWrapper SelectBase__field"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="searchbox"
              tabIndex={0}
            >
              <div
                className="FieldWrapper__content"
              >
                <div
                  className="AutosuggestMulti__wrapper"
                >
                  <input
                    aria-activedescendant={null}
                    aria-autocomplete="list"
                    aria-controls={null}
                    aria-labelledby="downshift-0-label"
                    autoComplete="off"
                    className="AutosuggestMulti__input"
                    id="downshift-0-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="type here..."
                    value=""
                  />
                </div>
                <List
                  aria-labelledby="downshift-0-label"
                  doSelectOnNavigate={false}
                  id="downshift-0-menu"
                  isControlledNavigation={true}
                  isDivided={false}
                  role="listbox"
                >
                  <ul
                    aria-labelledby="downshift-0-label"
                    className="List"
                    id="downshift-0-menu"
                    role="listbox"
                  />
                </List>
              </div>
            </div>
          </FieldWrapper>
        </div>
      </Downshift>
    </div>
  </SelectBase>
</AutosuggestMulti>
`;

exports[`AutosuggestMulti rendering should render empty component correctly when focused 1`] = `
<AutosuggestMulti
  inputPlaceholder="type here..."
  onBlur={[MockFunction]}
  onInputValueChange={[MockFunction]}
  onSelectionChange={[MockFunction]}
  selectedSuggestions={Array []}
  suggestionToString={[Function]}
  suggestions={Array []}
>
  <SelectBase
    blurredRenderer={[Function]}
    clearInputAfterSelection={true}
    clearTitle=""
    focusedRenderer={[Function]}
    inputRef={
      Object {
        "current": <input
          aria-activedescendant="downshift-1-item-0"
          aria-autocomplete="list"
          aria-controls="downshift-1-menu"
          aria-labelledby="downshift-1-label"
          autocomplete="off"
          class="AutosuggestMulti__input"
          id="downshift-1-input"
          placeholder="type here..."
          value=""
        />,
      }
    }
    keepExpandedAfterSelection={true}
    listRenderer={[Function]}
    onBlur={[MockFunction]}
    onInputValueChange={[Function]}
    onSelectionChange={[MockFunction]}
    showClearButton={false}
    suggestionToString={[Function]}
    suggestions={Array []}
  >
    <div
      className="SelectBase"
    >
      <Downshift
        defaultHighlightedIndex={0}
        defaultIsOpen={false}
        environment={[Window]}
        getA11yStatusMessage={[Function]}
        inputValue=""
        itemToString={[Function]}
        onChange={[Function]}
        onInputValueChange={[Function]}
        onOuterClick={[Function]}
        onSelect={[Function]}
        onStateChange={[Function]}
        onUserAction={[Function]}
        scrollIntoView={[Function]}
        selectedItemChanged={[Function]}
        stateReducer={[Function]}
        suppressRefError={false}
      >
        <div
          aria-expanded={true}
          aria-haspopup="listbox"
          aria-labelledby="downshift-1-label"
          aria-owns="downshift-1-menu"
          className="SelectBase__main SelectBase__main--focused"
          role="combobox"
        >
          <FieldWrapper
            className="SelectBase__field"
            clearLabel=""
            isFocused={true}
            onClear={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            showClearButton={false}
          >
            <div
              className="FieldWrapper FieldWrapper--isFocused SelectBase__field"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="searchbox"
              tabIndex={0}
            >
              <div
                className="FieldWrapper__content"
              >
                <div
                  className="AutosuggestMulti__wrapper AutosuggestMulti__wrapper--isFocused"
                >
                  <input
                    aria-activedescendant="downshift-1-item-0"
                    aria-autocomplete="list"
                    aria-controls="downshift-1-menu"
                    aria-labelledby="downshift-1-label"
                    autoComplete="off"
                    className="AutosuggestMulti__input"
                    id="downshift-1-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="type here..."
                    value=""
                  />
                </div>
                <List
                  aria-labelledby="downshift-1-label"
                  className="SelectBase__list--focused"
                  doSelectOnNavigate={false}
                  id="downshift-1-menu"
                  isControlledNavigation={true}
                  isDivided={false}
                  role="listbox"
                >
                  <ul
                    aria-labelledby="downshift-1-label"
                    className="List SelectBase__list--focused"
                    id="downshift-1-menu"
                    role="listbox"
                  >
                    <SuggestionsList
                      className="List__item"
                      getItemProps={[Function]}
                      highlightedIndex={0}
                      inputValue=""
                      key=".0"
                      noSuggestionsPlaceholder=""
                      suggestionToString={[Function]}
                      suggestions={Array []}
                      useOptimizeRender={false}
                    />
                  </ul>
                </List>
              </div>
            </div>
          </FieldWrapper>
        </div>
      </Downshift>
    </div>
  </SelectBase>
</AutosuggestMulti>
`;
